nginx/fulltest-provider/common.py

```diff
@@ -0,0 +1,154 @@
+import os
+import subprocess
+import pathlib
+import psutil
+import time
+
+key_exchanges = [
+##### OQS_TEMPLATE_FRAGMENT_KEX_ALGS_START
```

Comment:
Really confused by this: We're running `jinja2`in oqs-demos? Which script does this? I'd _really_ refrain from this complication. Why not query openssl for the algorithms?

Reply:
Ahh -- now I get it: You copied this file over from https://github.com/open-quantum-safe/oqs-provider/pull/173, right? IMO this is confusing: Please at least do it as before and pull it over via `wget` (such that one knows where it's coming from -- and that it's somewhat current as `jinja2` only runs in `oqs-provider`) -- or (my strong preference) please do it so that it extracts algorithms from running `openssl list` and thus adapts fully to the then-current algorithm capabilities in `oqs-provider` (whichever binary one chooses to run).

Reply:
> Ahh -- now I get it: You copied this file over from [open-quantum-safe/oqs-provider#173](https://github.com/open-quantum-safe/oqs-provider/pull/173), right? IMO this is confusing: Please at least do it as before and pull it over via `wget` (such that one knows where it's coming from -- and that it's somewhat current as `jinja2` only runs in `oqs-provider`)

Yes exactly, I intended to uncomment https://github.com/open-quantum-safe/oqs-demos/blob/ed50ec659841b4a26a02fcdc6ccb7abe0b62eb84/nginx/fulltest-provider/build_ubuntu.sh#L5-L6
once the script is merged in oqs-provider, and then delete it here. Wget'ting it now from the PR in oqs-provider would lead to a stale link once the PR is merged (or not).

> or (my strong preference) please do it so that it extracts algorithms from running `openssl list` and thus adapts fully to the then-current algorithm capabilities in `oqs-provider` (whichever binary one chooses to run).

I also need the security strengths to be able to generate a config that only consists of sigs and kems with matching strengths. I don't think this info is available with `openssl list`, or is it? Would be great if it is. If not, you see a way to get this info without jinja templating?

Reply:
> I don't think this info is available with openssl list, or is it?

It is: https://github.com/open-quantum-safe/oqs-provider/pull/156

Reply:
Oh, that's nice, thank you for the pointer! This really seems the better solution.

I'd then change this to use `openssl list` instead of jinja templating, remove the jinja-generated parts from the common.py and keep the rest of the script here. So https://github.com/open-quantum-safe/oqs-provider/pull/173 can be closed as well.

It will probably take me a few days to come back to this here.

Reply:
The PR is updated with a commit that removes jinja-generated parts and directly gets the algorithms, including their security strengths with `openssl list`.